primo/opt_model/result_parser.py

```diff
@@ -0,0 +1,804 @@
+        return iter(self.well_data.data.index)
+
+    def __str__(self) -> str:
+        msg = (
+            f"Number of wells in project {self.project_id}\t\t: {self.num_wells}\n"
+            f"Estimated Project Cost\t\t\t: ${round(self.plugging_cost)}\n"
+            f"Impact Score [0-100]\t\t\t: {self.impact_score}\n"
+            f"Efficiency Score [0-100]\t\t: {self.efficiency_score}\n"
+        )
+        return msg
+
+    def _check_column_exists(self, col_name):
+        """
+        Checks if a column exists
+        """
+        if col_name is None:
+            raise ValueError(f"Information for column {col_name} is not available.")
+
+    @property
+    def num_wells_near_hospitals(self):
+        """Returns number of wells that are near hospitals"""
+        col_name = self._col_names.hospitals
+        self._check_column_exists(col_name)
+
+        return len(self.well_data.data[self.well_data.data[col_name] > 0].index)
+
+    @property
+    def num_wells_near_schools(self):
+        """Returns number of wells that are near schools"""
+        col_name = self._col_names.schools
+        self._check_column_exists(col_name)
+        return len(self.well_data.data[self.well_data.data[col_name] > 0].index)
+
+    @property
+    def average_age(self):
+        """
+        Returns average age of the wells in the project
+        """
+        col_name = self._col_names.age
+        self._check_column_exists(col_name)
+        return calculate_average(self.well_data.data, col_name)
+
+    @property
+    def age_range(self):
+        """
+        Returns the range of the age of the project
+        """
+        col_name = self._col_names.age
+        self._check_column_exists(col_name)
+        return calculate_range(self.well_data.data, col_name)
+
+    @property
+    def average_depth(self):
+        """
+        Returns the average depth of the project
+        """
+        col_name = self._col_names.depth
+        self._check_column_exists(col_name)
+        return calculate_average(self.well_data.data, col_name)
+
+    @property
+    def depth_range(self):
+        """
+        Returns the range of the depth of the project
+        """
+        col_name = self._col_names.depth
+        self._check_column_exists(col_name)
+        return calculate_range(self.well_data.data, col_name)
+
+    @property
+    def ave_elevation_delta(self):
+        """
+        Returns the average elevation delta of the project
+        """
+        if not hasattr(self._col_names, "elevation_delta"):
+            raise AttributeError("There is no data for the elevation delta")
+        col_name = self._col_names.elevation_delta
+        self._check_column_exists(col_name)
+        return calculate_average(self.well_data.data, col_name, estimation_method="yes")
+
+    @property
+    def centroid(self):
+        """
+        Returns the centroid of the project
+        """
+        self._check_column_exists(self._col_names.latitude)

```

Comment:
Latitude and longitude are essential columns (WellData object would not have been created without these columns), so we do not have to perform these checks again.

Reply:
Addressed.